todo/ios/MongoDBSample/Classes/Controllers/Authentication/AuthenticationViewController.swift

```diff
@@ -53,54 +56,35 @@ class AuthenticationViewController: UIViewController, GIDSignInUIDelegate, GIDSi
         authOptionsStackView.removeArrangedSubview(skipButton)
         
+        handleAuthenticationProviders()
     }
     
-    func onReady(_ stitchClient: StitchClient) {
-
-        self.stitchClient = stitchClient
-        // fetch auth providers
-        stitchClient.fetchAuthProviders().done { (authProviderInfo: AuthProviderInfo) in
-            
-            self.handleAuthenticationProviders(authInfo: authProviderInfo)
-           
-            }.catch { error in
-                print("failed to get Authentication Providers. error: \(error)")
-        }
-    }
-    
-    
     //MARK: - Helpers
     
-    func handleAuthenticationProviders(authInfo: AuthProviderInfo) {
+    func handleAuthenticationProviders() {
         
         DispatchQueue.main.async { [weak self] in
             
             guard let strongSelf = self else { return }
-            var authAvailable = false
             
-            if let googleAuthInfo = authInfo.googleProviderInfo {
-                authAvailable = true
-                
-                var configureError: NSError?
-                GGLContext.sharedInstance().configureWithError(&configureError)
-                assert(configureError == nil, "Error configuring Google services: \(configureError)")
-                
-                GIDSignIn.sharedInstance().clientID = "<iOS-application-client-id>"
-                GIDSignIn.sharedInstance().serverClientID = "<Web-application-client-id>"
-                GIDSignIn.sharedInstance().uiDelegate = self
-                GIDSignIn.sharedInstance().delegate = self
-                
-                GIDSignIn.sharedInstance().scopes = googleAuthInfo.metadataFields?.map { $0.name }
-                strongSelf.googleSignInButton.isHidden = false
-                strongSelf.authOptionsStackView.addArrangedSubview(strongSelf.googleSignInButton)
-            }
-            else {
-                strongSelf.googleSignInButton.removeFromSuperview()
-            }
+            // Google authentication
+            GIDSignIn.sharedInstance().clientID = "<iOS-application-client-id>"
+            GIDSignIn.sharedInstance().serverClientID = "<Web-application-client-id>"
+            GIDSignIn.sharedInstance().uiDelegate = self
+            GIDSignIn.sharedInstance().delegate = self
+
+            strongSelf.googleSignInButton.isHidden = false
+            strongSelf.authOptionsStackView.addArrangedSubview(strongSelf.googleSignInButton)
             
-            if let facebookAuthInfo = authInfo.facebookProviderInfo {
-                authAvailable = true
-                var readPermissions: [FacebookCore.ReadPermission] = []
+            // Facebook authentication
+            let fbAppId = Bundle.main.object(forInfoDictionaryKey: "FacebookAppID") as? String
+            if fbAppId != nil && fbAppId != "<Your-FacebookAppID>" {
+                if let token = AccessToken.current {
+                    self?.loginWithFacebook(token: token)
```

Comment:
you can replace `self?` with `strongSelf`

Reply:
Done